### ISSUE TITLE
[Keyboard Navigation – Howdy – Script]: After selecting “Cancel” button, focus is not moving to the triggered element in Script tab.

### ISSUE THREAD

**Issue by v-gasikk** [howdyai/howdy-website#18]
**User Experience:**
Users who rely on keyboards will affect to navigate, if focus is not moving to the triggered element after selecting “Cancel” button.

**Test Environment:**
OS: Win 10 Version 1809 (OS Build 17763.316)
URL: https://howdy.ai/
Browser: Edge

**Repro Steps:**
Step 1: Open the above URL in edge browser.
Step 2: Open this URL: https://hcl-szz1925.howdy.ai/dashboard in edge.
Step 3: “Dashboard” page gets opened.
Step 4: Navigate to “Scripts” tab and select it.
Step 5: Navigate to “Edit” button and select it.
Step 6: Navigate to “Cancel” button and select it.
Step 7: Observe whether focus is moving to the triggered element after selecting “Cancel” button or not.
Step 8: Similarly verify for “Clone, Delete and Create new Script” also.

**Actual Result:**
After selecting “Cancel” button, focus is not moving to the triggered element in Script tab.

**Expected Result:**
After selecting “Cancel” button, focus should move to the triggered element in Script tab.

**MAS Reference Link:**
[https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={a4bbd37d-ee11-485e-b5c8-0539f0ac94b2}](url)

[MAS3.2.1_Howdy_Cancle button.zip](https://github.com/howdyai/howdy-website/files/2973772/MAS3.2.1_Howdy_Cancle.button.zip)

**Comment by benbrown** [comment]
@awalia13  This would be very difficult to achieve, since these buttons lead to a new page and the state on the previous page is not captured. Do you have any suggestions as to how others have achieved a similar result?

**Comment by benbrown** [comment]
OK!  Clicking "Cancel" from the edit screen will now send the user back to the "Edit" button they previously clicked.

**Comment by v-gasikk** [comment]
Verified. Issue is Partially Fixed. Hence Re-activating the bug.

After selecting "Cancel" button, now focus is moving to Triggered element i.e. "Edit".
Similarly Clone and Delete buttons also, focus should move to the triggered element.

**Comment by benbrown** [comment]
I believe this is now addressed:

* Click delete, click "cancel" in dialog, focus returns to delete button
* click edit, click "Cancel", return to list, focus returns to edit button
* Click clone, cancel button is no longer present (cannot cancel a clone)

**Comment by awalia13** [comment]
Closed as it is working fine
